pkg/options/search_option.go

```diff
@@ -0,0 +1,159 @@
+package options
+
+import (
+  "bytes"
+  "encoding/json"
+  "errors"
+  "fmt"
+  "github.com/go-http-utils/headers"
+  "github.com/sidilabs/kishell/pkg/config"
+  "mime"
+  "net/http"
+  "text/template"
+  "time"
+)
+
+type SearchParams struct {
+  Index string
+  Size int32
+  WindowFilter string
+  Zone string
+  Clause string
+  Older int64
+  Newer int64
+}
+
+type ResponseData struct {
+  Payload map[string]interface {}
+}
+
+const (
+  kibanaVersionHeaderKey = "kbn-version"
+  postContentType = "application/x-ndjson"
+  esSearchPath = "/elasticsearch/_msearch"
+  matchAllClause = `{"match_all": {}}`
+  queryClauseTemplate = `{"query_string":{"query":"{{.Query}}","analyze_wildcard":true,"default_field":"*"}}`
+  payloadTemplate = `{"index":"{{.Index}}","ignore_unavailable":true,"preference":1569331617740}
+{"version":true,"size":{{.Size}},"sort":[{"{{.WindowFilter}}":{"order":"desc","unmapped_type":"boolean"}}],"_source":{"excludes":[]},"aggs":{"2":{"date_histogram":{"field":"{{.WindowFilter}}","interval":"3h","time_zone":"{{.Zone}}","min_doc_count":1}}},"stored_fields":["*"],"script_fields":{},"query":{"bool":{"must":[{{.Clause}},{"range":{"{{.WindowFilter}}":{"gte":{{.Newer}},"lte":{{.Older}},"format":"epoch_millis"}}}],"filter":[],"should":[],"must_not":[]}},"highlight":{"pre_tags":["@kibana-highlighted-field@"],"post_tags":["@/kibana-highlighted-field@"],"fields":{"*":{}},"fragment_size":2147483647},"timeout":"30000ms"}
+`
+)
+
+func (r *ResponseData) PrintAllSources() error {
+  for _, element := range r.Payload {
+    responses := element.([]interface {})
+    for _, item := range responses {
+      response := item.(map[string]interface {})
+      hits := response["hits"].(map[string]interface {})["hits"].([]interface {})
+      for _, hitItem := range hits {
+        hit := hitItem.(map[string]interface {})
+        source := hit["_source"]
+        asJson, err := json.Marshal(source)
+        if err != nil {
+          return err
+        }
+        fmt.Println(string(asJson))
+      }
+    }
+  }
+  return nil
+}
+
+func (s *SearchCmd) Run(ctx *Context) error {
+  err := ctx.ConfigFile.IsEmpty()
+  if err != nil {
+    return err
+  }
+  clause := matchAllClause
+  if len(s.Query) > 0 {
+    out, err := buildFromTemplate("query", queryClauseTemplate, s)
+    if err != nil {
+      return err
+    }
+    clause = out.String()
+  }
+
+  server := ctx.ConfigFile.Servers[ctx.ConfigFile.CurrentServer]
+  if len(s.Server) > 0 {
+   server = ctx.ConfigFile.Servers[s.Server]
```

Comment:
You could easily check Server existence by
```suggestion
   server, ok = ctx.ConfigFile.Servers[s.Server]
   if !ok {
       return fmt.Errorf("Server %s does not exist", s.Server)
   }
```